lms/djangoapps/grades/subsection_grade.py

```diff
@@ -101,46 +101,57 @@ class SubsectionGrade(SubsectionGradeBase):
     """
```

Comment:
**Note to reviewers:** The bulk of the changes in this file are to support factory class-methods for creating and reading subsection grades.  Hopefully, this makes the code easier to understand - and also eliminates side-effects in `_compute_block_score`.